### PR TITLE
added input variables for pins, humidity, and temp values

### PR DESCRIPTION
**Pull request by croessi**
Agreed to make the changes stepwise and bring it back to thz504.yaml. 
First step I would need is to make the hardcoded pins variable to test the changes in my setup.

### CODE REVIEW

File: yaml/thz504.yaml [hunk 1]
```diff
@@ -1,3 +1,12 @@
+defaults:
+  clk_pin: "GPIO14"
+  mosi_pin: "GPIO13"
+  miso_pin: "GPIO4"
+  cs_pin: "GPIO15"
+  evu_pin: "GPIO5"
+  entity_room_temperature: "sensor.durchschnittstemperatur_haus_ohne_keller"
```
Comment (kr0ner) [inline yaml/thz504.yaml:7]: Should we pass this down from the top-level yaml? What do you think? Is more of an overall config, right?

**Comment by mkaiser** [comment]
hey there, 

I am using the ESP32 internal CAN Controller (https://esphome.io/components/canbus.html?highlight=can#esp32-can-component). 

could you also "elegantly integrate" a variant for this?

btw. I have a Tecalor TTC07cool heatpump and still have some issues with the existing implementations. This is why I'm following your progress here with your "clean c++ code approach" with great enthusiasm :)

File: yaml/thz504.yaml (hunk 1 above)
Comment (kr0ner) [reply yaml/thz504.yaml:7]: ok, I was on the wrong track again ... this is exactly what you are doing ^^

**Comment by mkaiser** [comment]
hey, sorry for the delayed response.

> Should we pass this down from the top-level yaml? What do you think? Is more of an overall config, right?

With my request I was referring to this. So keeping one yaml file as top (passing down such configurations) and having the sub-yaml files just for sensor configuration. 


Is it possible to move this part from thz504.yaml


```
#########################################
#                                       #
#   CANbus configuration                #
#                                       #
#########################################
canbus:
  - platform: mcp2515
    id: my_mcp2515
    spi_id: McpSpi
    cs_pin: $cs_pin
    can_id: 680
    use_extended_id: false
    bit_rate: 20kbps
    on_frame:

``` 

somehow to the top level entity? So users only have to change one file with their specific HW setup and could adapt "common" files for the sensor / register accesses?

**Pull request merged**